### DISCUSSION TITLE
Why NEXTAUTH_URL is required in build time?

### DISCUSSION THREAD

**Discussion by yarinsa** [nextauthjs/next-auth#7641] [Help]
### Question 💬

I was working on building our Next.js docker image. 
The process took a while as I had to map different issues & tools. 
One of the things I saw is that NextAuth.js is using NEXTAUTH_URL during build time. 
Even tho I red the docs and understand the usage. I am not sure why this is required in build time. 
I followed the code and saw that it's required for the client side of Next.js. But not sure why this can't be injected before the script is getting delivered to the customer. 
I will be happy to open a feature request :) , Want to check with the maintainers if that sounds reasonable or am I missing something 

### How to reproduce ☕️

Run build without nextauth url :)

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
It should not be/depends on your code. cannot help without a reproduction though.

**Reply by yarinsa** [reply]
Cool , I will reproduce and update this comment
